doc/api/stream.md

```diff
@@ -1739,6 +1739,30 @@ run().catch(console.error);
 after the `callback` has been invoked. In the case of reuse of streams after
 failure, this can cause event listener leaks and swallowed errors.
 
+### `stream.Readable.batched(stream, [options])`
+
+* `stream`
+* `options`
+
+Converts a `Readable` into an async iterator which
```

Comment:
I am not sure if this should return an async iterator or a readable stream - I would intuitively expect a `Readable.batched` to return a readable stream (like Readable.from ).

Reply:
Yea, I'm unsure how the API should look to make sense.

Reply:
If `Readable.batched` returned a readable stream it would already be async-iterable and you could also pass it around. The cost is that it would be slower to create upfront.

If we could somehow lazily call `Readable.from` on the generator if it's not consumed as an async iterable that would _probably_ be a good way to not make this slower while still meeting expectations.

I am fine with this returning an async generator (I want to see a benchmark to see this feature actually helps with this impl + tests for sure though :])